### PR TITLE
Drupal 10 compatibility fixes

### PR DESCRIPTION
**Pull request by juho-lehmonen**
# [UHF-9822](https://helsinkisolutionoffice.atlassian.net/browse/UHF-9822)

D10 compatibility fixes and minor coding quality improvements.

## To install:
* If you're using the Demo site testing branch, you can ignore these steps
* Use composer to download this module on this specific branch:
   * `composer require drupal/helfi_ahjo:dev-drupal-10-compatibility`
* Make sure the module installs correctly: `drush en helfi_ahjo`

## To test:
* [x] Check that the module can be configured from: `/admin/config/ahjo` (ask Juho for the correct API keys and URLs)
  * Add 00001 or 02900 as the starting ID and 4 as the depth (9999 will take a really long time and might cause timeouts).  
* [x] Run the taxonomy term sync from the config form and make sure the terms are created correctly: /admin/structure/taxonomy/manage/sote_section/overview
  * [ ] Try getting more terms by changing the starting ID to 00400 and the depth to 3, if it crashes try running it again before lowering the depth to 2. 
* [x] Add the org chart paragraph to a page to check that it works
* [x] Run `drush cex` to make sure the API key is not exported.
  * Try changing the API key setting and running `drush cim`, the API key should stay the same.
* [x] Check that the code changes follow our standards

## Other PRs
* https://github.com/City-of-Helsinki/drupal-helfi-platform-test/pull/44

[UHF-9822]: https://helsinkisolutionoffice.atlassian.net/browse/UHF-9822?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -2,26 +2,23 @@ on:
   pull_request:
   push:
     branches:
-      - develop
+      - main
```
Comment (hyrsky) [inline .github/workflows/ci.yml:5]: TODO after this is merged: change main branch to main from https://github.com/City-of-Helsinki/drupal-module-helfi-ahjo-org-chart/settings.

File: helfi_ahjo.module [hunk 2]
```diff
@@ -97,3 +97,22 @@ function helfi_ahjo_preprocess_paragraph(&$variables) {
...
   }
 }
+
+/**
+ * Implements hook_config_ignore_settings_alter().
+ */
+function helfi_ahjo_config_ignore_settings_alter(array &$settings) {
+  // Chatleijuke is used in hammashoito chat in Sote.
+  // Genesyschat might be used in palvelukeskus later but
+  // the block configs already removed from other instances.
```
Comment (hyrsky) [inline helfi_ahjo.module]: This comment is not relevant for this module.

**Review by hyrsky — CHANGES_REQUESTED** [review]

File: helfi_ahjo.module
Comment (hyrsky) [inline helfi_ahjo.module:1]: I think this module needs to implement `hook_helfi_paragraph_types` for `sote_section` paragraph so it can be placed.

**Review by hyrsky — CHANGES_REQUESTED** [review]

File: helfi_ahjo.module (hunk 2 above)
Comment (hyrsky) [reply helfi_ahjo.module]: I don't think we need to add this modules config to config ignore at all, since we should set api keys in `*settings.php`.

Comment (juho-lehmonen) [reply helfi_ahjo.module]: This has been changed to only ignore the API key now.

**Review by hyrsky — APPROVED** [review]

**Pull request merged**